flycheck.el

```diff
@@ -5361,7 +5361,7 @@ the BUFFER that was checked respectively.
 See URL `https://palantir.github.io/tslint/' for more information
 about TSLint."
   (let* ((json-array-type 'list)
-         (tslint-json-output (json-read-from-string output))
+         (tslint-json-output (json-read-from-string (if (eq output "") "{}" output)))
```

Comment:
@thieman You can simplify this to `(and (not (string-empty-p output)) (json-read-from-string output))`.  That'll return `nil` if there's no output.